_posts/2018-10-11-migrating-from-galen-to-backstop.md

```diff
@@ -0,0 +1,242 @@
+---
+layout: post
+title:  "Migrating from Galen to Backstop"
+date: 2018-10-22 14:55:10 +0200
+categories: testing
+author_name: Jacek Smolak
+author_url : /author/jaceksmolak
+author_avatar: jaceksmolak
+read_time : 12
+show_related_posts: false
+square_related:
+feature_image: posts/2018-10-25-migrating-from-galen-to-backstop/migration.jpg
+---
+
+> Visual regression testing has recently started gaining popularity- and for good reason. At its most basic, it's a series of tests that run through your site, take screenshots of various components, compare those screenshots against a baseline, and alert you when things change.
+
+_(Css-tricks; [Visual Regression Testing with PhantomCSS](https://css-tricks.com/visual-regression-testing-with-phantomcss/); November 17,2015)_
+
+## The reason behind the change
+
+[HolidayCheck](https://www.holidaycheck.de/), as any other company, needs to run its business not worrying that some changes might break its product. One of the ways, of maintaining that peace of mind, is to ensure that visually everything is A-OK.
+
+How did we do that? Our project used [Galen](http://galenframework.com/) as a test runner. Tests were executed on virtual machines, provided by external partner. At the beginning everything worked smoothly (OK, perhaps not, but it was maintained by few nice engineers, so that the rest of us could use that tool). Yet, couple of months ago, things started to break badly. For the reasons, not clear to us, [Selenium](https://www.seleniumhq.org/) started failing, randomly, more and more often. Also, tests were being executed slower and slower. Up to the point where the whole CI pipeline took ~30min to finish. Yes, thirty.
+ 
+On top of that, when somebody merged something to the master branch, you had to rebase. That meant yet another 30min. What if this time the tests have failed? You would have to rerun them. What if something was merged again? You'd have to rebase, yet again, and pray, that the random fail won't happen. Some features took hours to get to production.
+ 
+Enough was enough.
+
+## Call for action
+
+The first thing we did, was removing some of the tests, the ones that were not important for the business to run. We managed to go down from 30 to 20 min. But that was not enough. And we did not remove anything regarding random failures. What we gained, though, was a list of tests that really mattered. And those were the ones we decided to migrate.
+
+The search for a replacement for Galen was rather short. We googled for some solutions, but we also checked our own backyard and found out that other team was using [Backstop](https://garris.github.io/BackstopJS/) for their own project. Decision was easy: let's not use yet another tool, creating yet another knowledge silo. Let's use something that is known and, more or less, proved to be working.
```

Comment:
Use `BackstopJS` for link label as it is the full name of the library

Reply:
Hmm ... do you think I should rename all occurrences of Backstop?

Reply:
No, its fine. Actually I've just checked that they sometimes referring to it as Backstop as well